Wikipedia/Code/WMFExploreSectionControllerCache.m

```diff
@@ -43,22 +50,80 @@ - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
         self.dataStore                              = dataStore;
         self.sectionControllersBySection            = [[NSCache alloc] init];
         self.sectionControllersBySection.countLimit = [WMFExploreSection totalMaxNumberOfSections];
-        self.sectionsBySectionController            = [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory | NSMapTableObjectPointerPersonality
-                                                                            valueOptions:NSMapTableWeakMemory];
+
+        /*
+         Controllers (keys in the map) are retained by the NSCache and can be evicted at any time. On the other hand,
+         sections (values in the map) are "value types" which should be owned by the receiver.  In other words, sections
+         are created as needed, and not retained for any particular reason, and it would be unintentional for a value in
```

Comment:
Basically, comes down to this:

> I'm looking at this from the other side. I can't think of a reason that the reverse map should ever retain anything **since it is just a convienence and not a cache itself.**

I wasn't really taking that to heart.  I'm against "weakly" referencing sections on principle, but in this case it doesn't matter since the weak map is just another "view" of the cache its associated with.  Put another way, the weak map only exists because it's not possible to do reverse lookup of `NSCache` since it doesn't expose its keys.

I'll change this on or before Monday.

---

> Not following. Retrieving a controller for a copy of a section is done by the cache, not by this object. Did you say this backwards?

Right, retrieving a section by its corresponding controller should always be possible even if nothing is referencing it.

> This seems like a separate issue from retaining. This has to do with the object personality setting, no?

Right, the **controllers** are compared using object personality, _not_ the sections.  Meaning, any two equivalent sections can be treated as if they're the same object, since they're immutable.  This is what I mean when I say sections are struct-like.

> we aren't in swift and nothing is a struct.

No, but "value types" are just a concept.  Swift structs are one implementation, doesn't mean we can't have value types in ObjC.